create_bot_knowledge.py

```diff
@@ -88,126 +134,52 @@ def extract_knowledge_from_repository(project: Tuple[str, str], update_knowledge
-            return 0
-
-    if pull_requests:
-
-        if update_knowledge:
-            results = data["results"]
-
-        else:
-            results = {}
-
-        pr_n = 1
-        for pr in pull_requests:
-            _LOGGER.info(f"Analyzing PR {pr_n}/{len(pull_requests)}")
-            _LOGGER.debug(f"PR ID: {pr.id}")
-            pull = repo.get_pull(pr.id)
-            _LOGGER.debug(f"PR commits number: {pull.commits}")
-            commits = pull.commits
-            # TODO: Use commits to extract information.
-            # commits = [commit for commit in pull.get_commits()]
-            reviews = pull.get_reviews()
-            labels = pull.get_labels()
-            author = str(pull.user.login)
-
-            # Consider time of approved PR
-            try:
-                step = 0
-                while reviews[step].state != "APPROVED":
-                    step += 1
-                pull_requested_approved = reviews[step].submitted_at
-                results[str(pr.id)] = {
-                    "PR_labels": [label.name for label in labels],
-                    "PR_created": pull.created_at.timestamp(),
-                    "PR_approved": pull_requested_approved.timestamp(),
-                    "PR_approved_by": str(reviews[step].user.login),
-                    "PR_TTR": pull_requested_approved.timestamp() - pull.created_at.timestamp(),
-                    "PR_author": author,
-                    "PR_commits_number": commits
-                }
-            except:
-                # No review has been submitted (e.g. automatically merged)
-                results[str(pr.id)] = {
-                    "PR_labels": [label.name for label in labels],
-                    "PR_created": pull.created_at.timestamp(),
-                    "PR_approved": None,
-                    "PR_approved_by": None,
-                    "PR_TTR": None,
-                    "PR_author": author,
-                    "PR_commits_number": commits
-                }
-                pass
-
-            pr_n += 1
-
-        current_path = Path().cwd()
-        source_knowledge_repo = current_path.joinpath("./Bot_Knowledge")
-        source_knowledge_file = source_knowledge_repo.joinpath(f'{project[1] + "-" + project[0]}.json')
-
-        if update_knowledge:
-            _LOGGER.info(f"Updating knowledge for: {project[1] + '/' + project[0]}")
-
-        project_results = {"name": project[1] + "/" + project[0], "results": results}
-        with open(source_knowledge_file, "w") as fp:
-            json.dump(project_results, fp)
-
-    else:
+    if not pull_requests:
         _LOGGER.info(f"No new knowledge from repo {project[1] + '/' + project[0]}")
-
-    return 0
+        return
+
+    results = data['results'] if update_knowledge else {}
+    
+    for pr_number, pr in enumerate(pull_requests, start=1):
+        pull = repo.get_pull(pr.id)
+        
+        _LOGGER.info(f"Analyzing PR {pr_number}/{len(pull_requests)}")
```

Comment:
How about `_LOGGER.info(f"Analyzing PR n. {pr_number}/{len(pull_requests)}")`? :)

Reply:
ah... @fridex is a big fan of not using f-strings with the logger ;)

Reply:
Right :P we need to reformat them all with `_LOGGER.info("Analyzing PR n. %r/%r" % pr_number, len(pull_requests))`.

Reply:
Done